rpc_util.go

```diff
@@ -570,7 +570,7 @@ func (p *parser) recvMsg(maxReceiveMessageSize int) (pf payloadFormat, msg []byt
 		return 0, nil, status.Errorf(codes.ResourceExhausted, "grpc: received message larger than max length allowed on current machine (%d vs. %d)", length, maxInt)
 	}
 	if int(length) > maxReceiveMessageSize {
-		return 0, nil, status.Errorf(codes.ResourceExhausted, "grpc: received message larger than max (%d vs. %d)", length, maxReceiveMessageSize)
+		return 0, nil, status.Errorf(codes.ResourceExhausted, "grpc: received message after decompression larger than max (%d vs. %d)", length, maxReceiveMessageSize)
```

Comment:
This isn't correct.  This is the raw message before decompression.  `recvAndDecompress` calls this and decompresses this payload as a separate step.

Reply:
Also note that we can't call this the "compressed" message, since we don't know whether it's compressed in the first place.  `pf` can tell us whether it's compressed if we can't come up with a generic enough message to convey the idea properly.

Reply:
@dfawley 
I have a small question, I was writing this fix after reading this comment here:

https://github.com/grpc/grpc-go/issues/4761#issuecomment-918817668

In this, the issue creator says that he is "compressing" an `8 MB` message to `4 MB` and then sending it over.
Now, in our logic, since we are decompressing it, later on, is it possible that your current logic and error used are absolutely right and the message was not compressed in the first place?

Reply:
We check the message size twice.  Here (not yet decompressed) and here after decompressing:

https://github.com/grpc/grpc-go/blob/3b94303f375441714f9f4c488781e8928e000537/rpc_util.go#L721

And if a message isn't compressed, it seems like we might be checking the size twice, which we could clean up as part of this.  The issue is about clarifying whether it's the compressed message or the decompressed message size was over the limit (if the message is compressed).